### PR TITLE
Fix Asn construction from BigInteger

### PR DESCRIPTION
**Pull request by lolepezy**
This to fix the bug when a very big `BigInteger` can be used for constructing an `Asn` silently wrapping around to a `long`.

### CODE REVIEW

File: src/test/java/net/ripe/ipresource/AsnTest.java [hunk 1]
```diff
@@ -80,6 +82,13 @@ public void shouldFailOnIllegalValue() {
         Asn.parse("AS232442321412");
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldFailOnWraparoundBigInteger() {
+        // BigInteger.longValue() keeps only the low 64 bits, so converting this value with longValue()
+        // would wrap around to 5, which is in range. Asn(BigInteger) must reject it regardless.
+        new Asn(BigInteger.ONE.shiftLeft(64).add(BigInteger.valueOf(5)));
```
Comment (bjpbakker) [inline src/test/java/net/ripe/ipresource/AsnTest.java:94]: The implemented checks use the AS domain boundaries, not the `longValue` boundaries. Consider adding specific tests for these AS boundaries.

File: src/main/java/net/ripe/ipresource/Asn.java [hunk 2]
```diff
@@ -54,8 +54,8 @@ public class Asn extends UniqueIpResource {
     // unsigned long value.
     private int intValue;
 
-    public Asn(BigInteger value) {
-        this(value.longValue());
+    public Asn(BigInteger value) {        
+        this(checked(value));
```
Comment (bjpbakker) [inline src/main/java/net/ripe/ipresource/Asn.java]: Domain boundaries are already checked in the `Asn(long)` constructor. Instead of checking them twice, using `this(value.longValueExact())` is more useful IMO. This throws when the BigInteger value overflows a `long`.

**Review by bjpbakker — CHANGES_REQUESTED** [review]
